_posts/2023-03-09-solidity-developer-survey-2022-results.md

```diff
@@ -0,0 +1,451 @@
+
+![Time to Productiveness](/img/2023/02/sol_productivity.png)
+
+#### Solidity User Profile and Usage Habits
+
+With regards to usage frequency, more than 40% of respondents use Solidity on a daily basis! 37.3% use it weekly, and 13.9% on a monthly basis. 
+
+Roughly 8% indicated to be using Solidity "rarely" or "never". Most of them indicated before that they use Solidity for personal projects and code in a different programming language at work.
+
+![Solidity Usage Frequency](/img/2023/02/sol_frequency.png)
+
+A striking 81.8% of all respondents use Visual Studio Code as their editor when writing Solidity code. Vim and IntelliJ follow in the second and third ranks with 3.7 and 3.4% usage, respectively.
+
+Compared to the previous survey in 2021, Visual Studio code gained significantly in popularity (from roughly 50% to 81.8%).
+
+![Editor Overview](/img/2023/02/editor.png)
+
+Depending on the chosen IDE, we also asked respondents which Solidity-related plugins they use, if any. “HardHat VSCode” by Nomic Foundation and the “Solidity” extension by Juan Blanco (both for Visual Studio Code) are the most popular.
+
+![Editor Plugins Overview](/img/2023/02/editor_plugins.png)
+
+Hardhat remains the most popular Ethereum-specific development environment, with roughly 75% of all respondents using Hardhat.
+
+Remix follows with 42%. Foundry has significantly increased its share from 1.6% in 2021 to 30% in 2022.
+
+Truffle continues to move more into the background, with 17% of respondents indicating that they use it.
+
+Rather "niche" Ethereum-specific development environments are Brownie (6.7%), Ape (3.3%), Dapptools (2.3%), and Embark (0.8%). 
+
+4.4% of respondents are not using any Ethereum-specific development environment.
+
+It’s worth noting that this question was a checkbox question, allowing respondents to select multiple answers.
+
+_⚠️ Comparing the results from 2020, 2021, to 2022 may offer some insights like Truffle losing a significant share (2020: 59.3% -> 2021: 26.2% -> 2022: 17%), while Hardhat, and newcomers like Foundry increased their share in users. However, it's important to consider that the previous surveys had significantly fewer responses (2020: 194, 2021: 435, 2022: 1401). A year-on-year comparison can only be interpreted as a loose trend and it’s not the intent of this study to analyze user splits between IDEs in detail._
+
+![Ethereum IDE Overview](/img/2023/02/eth_ide.png)
+
+With roughly 90%, 0.8.x Solidity versions remain to be the by far most used ones. The usage share of both the 0.7.x (10.2%) and the 0.6.x (7.7%) series continues to decrease since the previous survey. Everything older than that is hardly in use anymore.
+
+![Used Solidity Versions](/img/2023/02/versions.png)
+
+_⚠️ Reminder: Please make sure to frequently update your code. [Several important bug fixes and security improvements](https://github.com/ethereum/solidity/blob/develop/docs/bugs_by_version.json) are added in the newer versions!_
```

Comment:
update your code and/or compiler?

Reply:
We could say "update your code to the latest Solidity version", which implies having an updated compiler?

Reply:
not necessarily.. I guess this would only be the case with breaking releases.. for non breaking releases there is theoretically no need to update the code, but updating the compiler could have benefits.. I guess it cannot hurt to explicitly write somewhere that the latest solidity compiler should always be used